Sources/ApodiniREST/RESTInterfaceExporter.swift

```diff
@@ -8,32 +8,94 @@ import NIO
+/// Public Apodini Interface Exporter for REST
+public final class RESTInterfaceExporter: Configuration {
+    let configuration: RESTExporterConfiguration
+    var staticConfigurations: [RESTDependentStaticConfiguration]
+    
+    /// The default `AnyEncoder`, a `JSONEncoder` with certain set parameters
+    public static var defaultEncoder: AnyEncoder {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.prettyPrinted, .withoutEscapingSlashes]
+        return encoder
+    }
+    
+    /// The default `AnyDecoder`, a `JSONDecoder`
+    public static var defaultDecoder: AnyDecoder {
+        JSONDecoder()
+    }
+    
+    /**
+     Initializes the configuration of the `RESTInterfaceExporter` with (default) `AnyEncoder` and `AnyDecoder`
+     - Parameters:
+         - encoder: The to be used `AnyEncoder`, defaults to a `JSONEncoder`
+         - decoder: The to be used `AnyDecoder`, defaults to a `JSONDecoder`
+     */
+    public init(encoder: AnyEncoder = defaultEncoder, decoder: AnyDecoder = defaultDecoder) {
+        self.configuration = RESTExporterConfiguration(encoder: encoder, decoder: decoder)
+        self.staticConfigurations = [EmptyRESTDependentStaticConfiguration()]
+    }
+    
+    public func configure(_ app: Apodini.Application) {
+        /// Instanciate exporter
+        let restExporter = _RESTInterfaceExporter(app, self.configuration)
+        
+        /// Insert exporter into `SemanticModelBuilder`
+        let builder = app.exporters.semanticModelBuilderBuilder
+        app.exporters.semanticModelBuilderBuilder = { model in
+            builder(model).with(exporter: restExporter)
+        }
+        
+        /// Configure attached related static configurations
+        self.staticConfigurations.configure(app, parentConfiguration: self.configuration)
+    }
+}
 
-    let app: Vapor.Application
-    let configuration: RESTConfiguration
+extension RESTInterfaceExporter {
+    /**
+     Initializes the configuration of the `RESTInterfaceExporter` with (default) JSON Coders and possibly associated Exporters (eg. OpenAPI Exporter)
+     - Parameters:
+         - encoder: The to be used `JSONEncoder`, defaults to a `JSONEncoder`
+         - decoder: The to be used `JSONDecoder`, defaults to a `JSONDecoder`
+         - staticConfiguraiton: A result builder that allows passing dependend static Exporters like the OpenAPI Exporter
+     */
+    public convenience init(encoder: JSONEncoder = defaultEncoder as! JSONEncoder,
+                            decoder: JSONDecoder = defaultDecoder as! JSONDecoder,
+                            @RESTDependentStaticConfigurationBuilder staticConfigurations: () -> [RESTDependentStaticConfiguration] = { [] }) {
+        self.init(encoder: encoder, decoder: decoder)
+        self.staticConfigurations = staticConfigurations()
+    }
+}
 
+/// Internal Apodini Interface Exporter for REST
+// swiftlint:disable type_name
+final class _RESTInterfaceExporter: InterfaceExporter {
```

Comment:
To be honest I am not too much of a fan of the underscores being added to all exporters. Maybe a more pragmatic way: What about naming the configuration `REST`? I think that the average Apodini user has no clue what an InterfaceExporter is anyways and just naming the configuration `REST` probably makes it quite easy to find. Same applies to `OpenAPI`, `GRPC` and so on ...